src/API/Google/AdsCampaign.php

```diff
@@ -96,10 +97,13 @@ public function get_campaigns(): array {
 		} catch ( ApiException $e ) {
 			do_action( 'woocommerce_gla_ads_client_exception', $e, __METHOD__ );
 
-			throw new Exception(
+			$errors = $this->get_api_exception_errors( $e );
+			throw new ExceptionWithResponseData(
 				/* translators: %s Error message */
-				sprintf( __( 'Error retrieving campaigns: %s', 'google-listings-and-ads' ), $e->getBasicMessage() ),
-				$this->map_grpc_code_to_http_status_code( $e )
+				sprintf( __( 'Error retrieving campaigns: %s', 'google-listings-and-ads' ), reset( $errors ) ),
+				$this->map_grpc_code_to_http_status_code( $e ),
```

Comment:
❓ 

- Why we are attaching the message of the first error in the Exception message. Feels now redundant for me since the exception could have multiple errors that are now attached in the data fragment. 
- What if the first error is different for the old `$e->getBasicMessage()`. Are we sure some kind of frontend work is not relying on that message due to the lack of error status codes until now?

If there is no other reason I'd mantain the previous line of 

`sprintf( __( 'Error retrieving campaigns: %s', 'google-listings-and-ads' ), $e->getBasicMessage() ),`

Reply:
The frontend responds only to the error code so the message is ignored, after these changes the errors can be used but that's not decided yet, because they won't be translatable (coming straight from the API they will always be english). The reason why I changed it to the first message, is because that's the one with the most descriptive error message.
If we default back to the basic message we will get "Request contains an invalid argument" for almost every error, which doesn't really tell you anything.

Reply:
Thanks for the explanation. 

Not sure about the benefit of overriding the message if it's going to be ignored. However since it seems to be under discussion we can go ahead and modify it in the future if needed. 

> If we default back to the basic message we will get "Request contains an invalid argument" for almost every error, which doesn't really tell you anything.

❓  "Request contains an invalid argument" is this something returned by Google (rpc)? 

My thoughts are about other cases in which `$e->getBasicMessage()` could provide for us a better message than the first item in the `$errors` array, or even a message that is different from the `$errors` array.

Reply:
I haven't found any other cases where `$e->getBasicMessage()` returns a better error message. We use a very limited part of the Ads API, and I've pretty extensively tested a lot of error scenarios. If there aren't any detailed error messages it will still revert back to the basic message.

The basic message is from the ApiException object, but the detailed messages are from within a GoogleAdsFailure object. Although I haven't found any straightforward helper functions to extract that, but the GoogleAdsFailure is what we ideally want to be working with since it contains the actual error messages.